src/document.c

```diff
@@ -1656,6 +1655,12 @@ gboolean document_reload_prompt(GeanyDocument *doc, const gchar *forced_enc)
 	g_free(base_name);
 	return result;
 }
+/* also used for reloading when forced_enc is NULL */
+gboolean document_rename_prompt(GeanyDocument *doc) {
```

Comment:
I don't think this convenience function should be added.  Not only it's very simple and only use "public" API, but it's also only used in one place.